### DISCUSSION TITLE
build apache2 with poky (or openembedded-core), target tegra-18x

### DISCUSSION THREAD

**Discussion by yoowonsuk** [OE4T/meta-tegra#1154] [Q&A]
Hello
I tried to build apache2 targeting jetson-agx-orin-devkit.conf only to fail
![image](https://user-images.githubusercontent.com/52575749/215395237-cc0ca402-2a23-4f8d-a82f-fa736f8c7a97.png)

I did,
1. clone poky first and then meta-tegra (i did openembedded-core instaed of poky but same)
2. add meta-tegra to bblayers.conf
3. change machine to jetson-agx-orin-devkit in local.conf

so as to get tegra-t18x target executive binary.
how could i progress next?

Thank you

**Comment by ichergui** [comment]
Hey @yoowonsuk 

Which branch you are using ? 
I quilt lost because you are mentioning `Jetson Orin AGX` and `Tegra-t18x` in you post. This is completely different Tegra family

**Comment by madisongh** [comment]
Based on the version numbers of the recipes involved, it looks like this is the `master` branch.  There have been some recent changes in bitbake and OE-Core around parsing and the bitbake data store that could cause errors like those you mention; make sure you have updated both of those repos to latest.

Or better yet, switch to a branch which is likely to be more stable, such as `kirkstone`.

**Comment by yoowonsuk** [comment]
I think Both are right

@ichergui
I used master branch and thought machine=tegra-t18x equals Jetson Orin AGX
I'm sorry to say, do you know where tegra-t18x recipe is? I can't find meta-nvidia or so on..

@madisongh 
I tried the newest only to fail.
one question, how to associate master with kirkstone?
I finally succeed it once, but i don't know what should be retouched right now.. (I just fixed this and that)

Thank you for help xD

**Reply by madisongh** [reply]
Sorry, I'm fixing the recipes up in real time here.  I think I've caught them all now, so updating to latest again in meta-tegra should at least get through parsing successfully.

I'm not sure what you mean by "associate master with kirkstone".  The convention that most meta-* layers follow is to have a branch that is named the same (or derived from) the corresponding branch in OE-Core.

As for "tegra-t18x", it's not clear what you're asking for.  There is no such machine defined in our layer; "t18x" is a generic term to refer to the Jetson TX2 family.  The specific machines support in meta-tegra can be found in the `meta-tegra/conf/machine` directory; the names of the configuration files there correspond to the configuration names NVIDIA uses for the Jetson development kits in its L4T/Jetson Linux.

For information on which of the branches in this repo support which versions of L4T and which specific machines, see [this wiki page](https://github.com/OE4T/meta-tegra/wiki/Which-branch-should-I-use%3F).

**Reply by yoowonsuk** [reply]
@ichergui @madisongh Thank you all!
I feel like I understand a little bit, since I'm not accustomed to it.
but yes, I should have set the same branch to OE-core one.

One thing to be remained is you said t18x is as to Jetson's family naming, also I think the name of "tegra" is in meta-tegra so isn't tegra-t18x related to your layers?

when searching, I found that tegra is a kind of nvidia chipset as well as t18x is sort of versioning.

I would much appreciate your letting me know where tegra-t18x machine recipe is. again sorry to force business on you.
And I will show you some.. xD it sounds like opensource, I guess
![image](https://user-images.githubusercontent.com/52575749/215668097-2c033804-a99e-4a54-913b-c40496c7852e.png)

**Reply by madisongh** [reply]
As I said, you'll find the machine configurations supported in meta-tegra in the [conf/machine](https://github.com/OE4T/meta-tegra/tree/master/conf/machine) directory in the layer.  I don't know where that one you screen-capped came from, so I assume it's another layer, or someone else's fork of this repository.  From the content of the screen cap, it looks like it's structured differently, and incompatibly, with our layer. I'd recommend contacting whoever you got that from for more information.

**Reply by yoowonsuk** [reply]
I should have known better than to check tegra-t18x first
Thank you for helping me!
